### PR TITLE
Persistence tweaks

### PR DESCRIPTION
**Pull request by Gaming32**
This PR removes unnecessary data from persisting. Now only data needed to recreate the current board is stored, and not full goal data. An upgrade path was added to convert data from the old format to the new. Additionally, *all* games are now persistent.

### CODE REVIEW

File: common/src/main/java/io/github/gaming32/bingo/util/BingoStreamCodecs.java [hunk 1]
```diff
@@ -16,4 +16,8 @@ public static <E extends Enum<E>> StreamCodec<FriendlyByteBuf, E> enum_(Class<E>
     public static <B extends ByteBuf, V> StreamCodec.CodecOperation<B, V, V[]> array(IntFunction<V[]> factory) {
         return codec -> codec.apply(ByteBufCodecs.list()).map(l -> l.toArray(factory), Arrays::asList);
     }
+
+    public static <B, V> StreamCodec<B, V> uncheckedUnit(V value) {
+        return StreamCodec.of((buf, ignored) -> {}, buf -> value);
```
Comment (Earthcomputer) [inline common/src/main/java/io/github/gaming32/bingo/util/BingoStreamCodecs.java:21]: What's the difference between this and `StreamCodec.unit`?

File: common/src/main/java/io/github/gaming32/bingo/BingoCommand.java [hunk 2]
```diff
@@ -240,7 +237,7 @@ public Component getDisplayName() {
                         final StringBuilder line = new StringBuilder(board.getSize());
                         for (int y = 0; y < board.getSize(); y++) {
                             for (int x = 0; x < board.getSize(); x++) {
-                                line.append(board.getGoal(x, y).goal().goal().getDifficulty().value().number());
+                                line.append(board.getGoal(x, y).difficulty().orElseThrow().value().number());
```
[the remark this answers is not in the document]
Comment (Gaming32) [reply common/src/main/java/io/github/gaming32/bingo/BingoCommand.java:240]: The difficulty is absent on the client.

File: common/src/main/java/io/github/gaming32/bingo/util/BingoStreamCodecs.java (hunk 1 above)
Comment (Gaming32) [reply common/src/main/java/io/github/gaming32/bingo/util/BingoStreamCodecs.java:21]: `StreamCodec.unit` checks that the value it's encoding is equal to the passed in value.

Comment (Earthcomputer) [reply common/src/main/java/io/github/gaming32/bingo/util/BingoStreamCodecs.java:21]: Ah, maybe that difference should be documented

**Review by Earthcomputer — APPROVED** [review]

**Pull request merged**